File: noble-idl-compiler/tests/compile_output/scala/sjs/MyExtern.scala
package dev.argon.nobidl.sjs.test

import dev.argon.nobidl.sjs.core.JSTypeAdapter

final class MyExtern
object MyExtern {
  def jsTypeAdapter: JSTypeAdapter[dev.argon.nobidl.scala_zio.test.MyExtern, MyExtern] =
    new JSTypeAdapter[dev.argon.nobidl.scala_zio.test.MyExtern, MyExtern] {
      override def toJS(a: dev.argon.nobidl.scala_zio.test.MyExtern): MyExtern = MyExtern()
      override def fromJS(b: MyExtern): dev.argon.nobidl.scala_zio.test.MyExtern = dev.argon.nobidl.scala_zio.test.MyExtern()
    }
}

final class MyExtern2[A]

File: langs/java/test/src/main/java/dev/argon/nobleidl/test/MyExtern.java
package dev.argon.nobleidl.example;

import dev.argon.nobleidl.runtime.graaljsInterop.JSAdapter;
import dev.argon.nobleidl.runtime.graaljsInterop.JSExecutor;

public record MyExtern() {
	public static JSAdapter<MyExtern> jsAdapter() {
		return new JSAdapter<>() {
			@Override
			public MyExtern fromJS(org.graalvm.polyglot.Context context, JSExecutor executor, org.graalvm.polyglot.Value value) {
				return new MyExtern();
			}

			@Override
			public org.graalvm.polyglot.Value toJS(org.graalvm.polyglot.Context context, JSExecutor executor, MyExtern value) {
				return context.eval("js", "({})");
			}
		};
	}
}
